core/templates/pages/exploration-editor-page/exploration-title-editor/exploration-title-editor.component.ts

```diff
@@ -29,9 +30,21 @@ angular.module('oppia').component('explorationTitleEditor', {
   },
   template: require('./exploration-title-editor.component.html'),
   controller: [
-    '$scope', 'ExplorationTitleService',
-    function($scope, ExplorationTitleService) {
+    '$rootScope', '$scope', 'ExplorationTitleService', 'FocusManagerService',
+    'RouterService',
+    function(
+        $rootScope, $scope, ExplorationTitleService, FocusManagerService,
+        RouterService) {
       $scope.explorationTitleService = ExplorationTitleService;
-    }
-  ]
+      var ctrl = this;
+      ctrl.$onInit = function() {
+        $rootScope.$watch(
```

Comment:
Ditto: Avoid $watch

Reply:
I was unable to find an alternative for this one. It was turning a bit ugly with other approaches. As it is a editor that loads inside settings tab it is a bit difficult to apply focus on it using event emitters inside the routing service. Except this one at all other places $watch is removed.

Reply:
Looking at the code, it seems like we seem to be "watching" the URL in the router service and on change we update the `activeTabName` variable. Can we do an emit whenever the tab name has changed and subscribe to this event in this component?

https://github.com/oppia/oppia/blob/e5a1ff215388707f4c00dcbc347b0bd7f2a5faea/core/templates/pages/exploration-editor-page/services/router.service.ts#L86-L150